src/garage/sampler/base.py

```diff
@@ -1,43 +1,71 @@
+"""Base sampler class."""
+
 import abc
+import copy
 
 
 class Sampler(abc.ABC):
-    @abc.abstractmethod
+    """Base class of all samplers."""
+
+    def __init__(self, algo, env):
+        """Construct a Sampler from an Algorithm.
+
+        Calling this method is deprecated.
+        """
+        self.algo = algo
+        self.env = env
+
+    @classmethod
+    def construct(cls, config, agent, env):
```

Comment:
do you envision replacing the constructor with this after the API change is finished?

Reply:
Maybe? Having the constructor be part of the API is generally avoided in languages where types aren't first class (since it's not meaningful there), so it's considered somewhat unusual in other languages. However, it seems fairly natural in Python. I don't have a strong preference here.